velox/functions/lib/ArrayShuffle.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+
+std::vector<std::shared_ptr<exec::FunctionSignature>>
+arrayShuffleWithSignatures();
+
+// This function returns metadata with 'deterministic' as false, it is used
+// with 'makeArrayShuffle'.
+exec::VectorFunctionMetadata getMetadataForArrayShuffle();
+
+// Shuffle with rand seed.
+std::shared_ptr<exec::VectorFunction> makeArrayShuffle(
+    const std::string& name,
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& config);
+
+std::vector<std::shared_ptr<exec::FunctionSignature>>
+arrayShuffleWithCustomSeedSignatures();
+
+// This function returns metadata with 'deterministic' as false, it is used
+// with 'makeArrayShuffleWithCustomSeed'.
+// This function needs to generate different sequences for the constant input,
+// set 'deterministic' as false to prevent the constant folding.
+exec::VectorFunctionMetadata getMetadataForArrayShuffleWithCustomSeed();
```

Comment:
Since both functions return the same values, let's consolidate into one: getMetadataForArrayShuffle